scripts/ensure.sh

```diff
@@ -1,14 +1,34 @@
 #!/bin/bash
 
-# Make sure the Node version matches what's in .nvmrc (and in our GitHub Actions workflows).
-current_version="$(node -v)"
-required_version="$(cat .nvmrc)"
+set -o errexit -o pipefail
 
-if [ ! $(echo ${current_version} | grep ${required_version}) ]; then
-    printf "\nIt looks like you're running Node %s, but this project uses Node %s.\n" ${current_version} ${required_version}
-    printf "If you're using nvm, try running nvm use.\n\n"
-    exit 1
-fi
+check_version() {
+    tool_name="$1"
+    executable="$2"
+    version_string="$(eval $3)"
+    required_version="$4"
+    level="$5"
+
+    color_red=$(echo -e "\033[0;31m")
+    color_yellow=$(echo -e "\033[0;33m")
+    color_end=$(echo -e "\033[0m")
+
+    details="See the README at https://github.com/pulumi/docs for a list of required tools and versions."
+
+    if ! command -v "$executable" &> /dev/null; then
+        echo -e "${color_red}error${color_end} This project requires $1, but the '$2' executable is either not installed or not on your PATH."
+        echo $details
+        exit 1
+    fi
+
+    if [ ! $(echo ${version_string} | grep ${required_version}) ]; then
+        printf "${color_yellow}warning${color_end} It looks like you're running %s %s, but this project uses version %s.\n" ${tool_name} ${version_string} ${required_version}
+        echo $details
+    fi
+}
+check_version "Node.js"  "node"  "node -v | sed 's/v\([0-9\.]*\).*$/\1/'"            "18"
+check_version "Hugo"     "hugo"  "hugo version | sed 's/hugo v\([0-9\.]*\).*$/\1/'"  "0.126.0"
+check_version "Yarn"     "yarn"  "yarn -v | sed 's/v\([0-9\.]*\).*$/\1/'"            "1.22"
```

Comment:
It'd be awesome if we could specify these versions in one place and use them everywhere. Don't yet have any great ideas for how to do this in a way that's work both here and in our GHA workflows, though.

Reply:
for `hugo` not sure, but I wouldn't be surprised if there is a way to enforce the node and yarn versions in the `package.json` file somehow.